lucet-runtime/lucet-runtime-tests/src/entrypoint.rs

```diff
@@ -119,14 +120,99 @@ pub fn mock_calculator_module() -> Arc<dyn Module> {
     }
 
     MockModuleBuilder::new()
-        .with_export_func(b"add_2", add_2 as *const extern "C" fn())
-        .with_export_func(b"add_10", add_10 as *const extern "C" fn())
-        .with_export_func(b"mul_2", mul_2 as *const extern "C" fn())
-        .with_export_func(b"add_f32_2", add_f32_2 as *const extern "C" fn())
-        .with_export_func(b"add_f64_2", add_f64_2 as *const extern "C" fn())
-        .with_export_func(b"add_f32_10", add_f32_10 as *const extern "C" fn())
-        .with_export_func(b"add_f64_10", add_f64_10 as *const extern "C" fn())
-        .with_export_func(b"add_mixed_20", add_mixed_20 as *const extern "C" fn())
+        .with_export_func(
+            b"add_2",
+            function_bytes_slice!(add_2),
+            &[],
+            Signature {
+                params: [ValueType::I64; 2].to_vec(),
+                ret_ty: Some(ValueType::I64),
+            },
+        )
+        .with_export_func(
+            b"add_10",
+            function_bytes_slice!(add_10),
+            &[],
+            Signature {
+                params: [ValueType::I64; 10].to_vec(),
+                ret_ty: Some(ValueType::I64),
+            },
+        )
+        .with_export_func(
+            b"mul_2",
+            function_bytes_slice!(mul_2),
+            &[],
+            Signature {
+                params: [ValueType::I64; 2].to_vec(),
+                ret_ty: Some(ValueType::I64),
+            },
+        )
+        .with_export_func(
+            b"add_f32_2",
+            function_bytes_slice!(add_f32_2),
+            &[],
+            Signature {
+                params: [ValueType::F32; 2].to_vec(),
+                ret_ty: Some(ValueType::F32),
+            },
+        )
+        .with_export_func(
+            b"add_f64_2",
+            function_bytes_slice!(add_f64_2),
+            &[],
+            Signature {
+                params: [ValueType::F64; 2].to_vec(),
+                ret_ty: Some(ValueType::F64),
+            },
+        )
+        .with_export_func(
+            b"add_f32_10",
+            function_bytes_slice!(add_f32_10),
+            &[],
+            Signature {
+                params: [ValueType::F32; 10].to_vec(),
+                ret_ty: Some(ValueType::F32),
+            },
+        )
+        .with_export_func(
+            b"add_f64_10",
+            function_bytes_slice!(add_f64_10),
+            &[],
+            Signature {
+                params: [ValueType::F64; 10].to_vec(),
+                ret_ty: Some(ValueType::F64),
+            },
+        )
+        .with_export_func(
+            b"add_mixed_20",
+            function_bytes_slice!(add_mixed_20),
+            &[],
+            Signature {
+                params: vec![
```

Comment:
a macro like this would be nice but I think a bit more time than I want to put in for the handful of uses at the moment, unless you know a nice way to expand `F64` and friends to `ValType::F64`? I adjusted `with_export_func` to take a builder that fills in defaults that makes this a little less tedious to work with, so that might help a bit

Reply:
The components of a path like that are separate tokens, so you should be able to use the ident matcher. Something like

```rust
( ( $($arg_ty:ident),* ) ) => { vec![ $($crate::ValType::$arg_ty),* ] }
```